x-pack/plugins/ml/public/application/app.tsx

```diff
@@ -27,6 +29,21 @@ interface AppProps {
 
 const localStorage = new Storage(window.localStorage);
 
+/**
+ * Provides global services available across the entire ML app.
+ */
+export function getMlGlobalServices(httpStart: HttpStart) {
+  const httpService = new HttpService(httpStart);
+  return {
+    httpService,
+    mlApiServices: mlApiServicesProvider(httpService),
+  };
+}
+
+export interface MlGlobalServices {
+  mlServices: ReturnType<typeof getMlGlobalServices>;
```

Comment:
Nit: Can we change this so the return type of `getMlGlobalServices` matches `MlGlobalServices`? Looking at the code I thought it was a bit strange that the names match but the shape of `MlGlobalServices` includes the wrapping `mlServices`. If `mlServices` was part of the returned object of `getMlGlobalServices` it could be used using the spread operator in the places where we get the service.

Reply:
Changed in fd9e7d0121c6ef58a5f535f9625e680c9db149c4